stock_product_set/models/product_set_line.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class ProductSetLine(models.Model):
+    _inherit = "product.set.line"
+
+    def prepare_picking_stock_move_values(self, picking, quantity):
+        self.ensure_one()
+        return {
+            "name": self.product_id.display_name,
+            "product_id": self.product_id.id,
+            "product_uom_qty": self.quantity * quantity,
+            "product_uom": self.product_id.uom_id.id,
+            # According to default values set from stock_move field in picking form
+            "location_id": picking.location_id.id,
+            "location_dest_id": picking.location_dest_id.id,
+            "group_id": picking.group_id.id,
+            "state": "draft",
+            "picking_id": picking.id,
+            "picking_type_id": picking.picking_type_id.id,
```

Comment:
You can use https://github.com/odoo/odoo/blob/260ef3dc73f5ed7d24f535145af01d7809dd7065/addons/purchase_stock/models/purchase.py#L506 as template.

I miss here at least `partner_id`.